InvenTree/stock/models.py

```diff
@@ -755,6 +756,14 @@ def get_part_name(self):
         related_name='build_outputs',
     )
 
+    consumed_by = models.ForeignKey(
+        'build.Build', on_delete=models.CASCADE,
```

Comment:
Question: Is it intended to delete the stock items completely when the build is deleted?

Reply:
Yes, that's correct.

The logic is:

1. Currently, untracked items are removed from the database (deleted) once consumed by a build
2. New behaviour is that the untracked items are marked as "consumed_by" - but kept in the database
3. The "consumed_by" field is what tells the database that these items are unavailable for consumption
4. When the build is deleted, I think the "cleanest" thing to do is to remove those items (as opposed to returning them to stock)

Open to alternative suggestions though :)